Improve navigation across all documentation (#74)
Dasharo/docs: docs/ecosystem/overview.md

* Open Source Firmware Community - semi-organised collections of contributors
  to projects related to embedded firmware. Dasharo project contributors are
  part of OSF community.

> * Development - Dasharo Development describe processes which Embedded Firmware

Reviewer: ```suggestion
* Development - Dasharo Development describes processes which Embedded Firmware
```

>   Team use for development of Dasharo Modules, which are sometimes custom

Reviewer: ```suggestion
  Team uses for development of Dasharo Modules, which are sometimes custom
```

>   Team uses for development of Dasharo Modules, which are sometimes custom

Reviewer 2: ```suggestion
  Team uses for the development of Dasharo Modules, which are sometimes custom
```

  services, but most of the time ready to use productised services, Open Core
  or SaaS.

> * Production - Dasharo Production describe process of putting together Dasharo

Reviewer: ```suggestion
* Production - Dasharo Production describes the process of putting together Dasharo
```

  Modules according to specific requirements. In this section we describe
  development infrastructure, build environment, CI/CD and build publishing

>   methods. We also cover how production interact with Dasharo Transparent

Reviewer: ```suggestion
  methods. We also cover how production interacts with Dasharo Transparent
```

  Validation System.

> * Customer Infrastructure - Some customers may have need of replicating part of

Reviewer 2: ```suggestion
* Customer Infrastructure - Some customers may have the need of replicating part of
```

>   Dasharo Production process and infrastructure in their own environment, what

Reviewer: ```suggestion
  Dasharo Production process and infrastructure in their own environment, which
```

  leads to repositories synchronization, different CI/CD setup and other needs.

>   This section describe interaction with Customer Infrastructure and what part

Reviewer: ```suggestion
  This section describes interaction with Customer Infrastructure and which part
```

>   of Dasharo Production is deployable on customer site.

Reviewer 2: ```suggestion
  of Dasharo Production is deployable on the customer site.
```

> * Transparent Validation - Dasharo Transparent Validation (DTV) is set of

Reviewer: ```suggestion
* Transparent Validation - Dasharo Transparent Validation (DTV) is a set of
```

  software, firmware and hardware components with the goal of simplifying and
  improving long term maintenance of Dasharo firmware. In this section we

>   introduce components, which belong to DTV.

Reviewer: ```suggestion
  introduce components which belong to DTV.
```

* Testing Infrastructure - Part of Dasharo Transparent Validation responsible
  for scheduling automated testing, publishing test results and interacting

>   with Dasharo Production in area of release candidate quality confirmation.

Reviewer 2: ```suggestion
  with Dasharo Production in the area of release candidate quality confirmation.
```
